BBallStatsTrack/src/com/example/bballstatstrack/activities/AddPlayersToTeamsActivity.java

```diff
@@ -0,0 +1,205 @@
+        return manager.findFragmentById( R.id.fragmentContainer );
+    }
+
+    private void replaceCurrentFragment( Fragment fragment )
+    {
+        FragmentManager manager = getFragmentManager();
+        FragmentTransaction transaction = manager.beginTransaction();
+        transaction.replace( R.id.fragmentContainer, fragment );
+        transaction.commit();
+    }
+
+    private void updateView( Fragment current )
+    {
+        if( mHomeTeamFragment.equals( current ) )
+        {
+            mHomeButton.setEnabled( false );
+            mAwayButton.setEnabled( true );
+        }
+        else if( mAwayTeamFragment.equals( current ) )
+        {
+            mHomeButton.setEnabled( true );
+            mAwayButton.setEnabled( false );
+        }
+        else
+        {
+            mHomeButton.setEnabled( true );
+            mAwayButton.setEnabled( true );
+        }
+    }
+
+    private class FragmentButtonListener implements OnClickListener
+    {
+        Fragment mFragment;
+
+        public FragmentButtonListener( Fragment fragment )
+        {
+            mFragment = fragment;
+        }
+
+        @Override
+        public void onClick( View v )
+        {
+            if( mFragment.equals( getCurrentFragment() ) )
+            {
+                return;
+            }
+            replaceCurrentFragment( mFragment );
+            updateView( mFragment );
+        }
+    }
+
+    private class ProceedButtonListener implements OnClickListener
+    {
+        public boolean hasCompleteTeamNames()
+        {
+            return mHomeName.isEmpty() || mAwayName.isEmpty();
+        }
+
+        public boolean hasInsufficientPlayers()
+        {
+            return mHomeTeam.size() < 5 || mAwayTeam.size() < 5;
+        }
+
+        @Override
+        public void onClick( View v )
+        {
+            mHomeTeam = mHomeTeamFragment.getPlayerList();
+            mHomeName = mHomeTeamFragment.getTeamName();
+            mAwayTeam = mAwayTeamFragment.getPlayerList();
+            mAwayName = mAwayTeamFragment.getTeamName();
+            if( hasInsufficientPlayers() )
+            {
+                Toast.makeText( AddPlayersToTeamsActivity.this, "Insuffecient players. Cannot proceed.",
+                        Toast.LENGTH_SHORT ).show();
+                return;
+            }
+            if( hasCompleteTeamNames() )
+            {
+
+                Toast.makeText( AddPlayersToTeamsActivity.this, "Team names are mandatory. Cannot proceed.",
+                        Toast.LENGTH_SHORT ).show();
+                return;
+            }
+            Intent intent = new Intent( AddPlayersToTeamsActivity.this, GameActivity.class );
+            intent.putExtra( HOME_TEAM_NAME, mHomeName );
+            setIntentExtras( intent, mHomeTeam, HOME_TEAM_NUMBERS, HOME_TEAM_MEMBER_NAMES );
+            intent.putExtra( AWAY_TEAM_NAME, mAwayName );
+            setIntentExtras( intent, mAwayTeam, AWAY_TEAM_NUMBERS, AWAY_TEAM_MEMBER_NAMES );
```

Comment:
mHomeTeam/mAwayTeam are, at this point, SparseArray<String>. The keys are used as player numbers, and the values are the players' names. The intent is passed on to GameActivity, which creates the Team objects.